src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseHybridFVMKernels.hpp

```diff
@@ -27,375 +27,346 @@
-                          arrayView1d< globalIndex const > const & faceDofNumber,
-                          arraySlice1d< localIndex const > const & elemToFaces,
-                          ElementViewConst< arrayView1d< real64 const > > const & mob,
-                          ElementViewConst< arrayView1d< real64 const > > const & dMob_dp,
-                          ElementViewConst< arrayView1d< globalIndex const > > const & elemDofNumber,
-                          real64 const (&oneSidedVolFlux)[ NF ],
-                          real64 const (&dOneSidedVolFlux_dp)[ NF ],
-                          real64 const (&dOneSidedVolFlux_dfp)[ NF ][ NF ],
-                          real64 const & dt,
-                          CRSMatrixView< real64, globalIndex const > const & localMatrix,
-                          arrayView1d< real64 > const & localRhs )
+  void computeFluxDivergence( localIndex const ei,
+                              StackVariables & stack ) const
   {
-    // fluxes
-    real64 divMassFluxes = 0;
-    real64 dDivMassFluxes_dElemVars[ NF+1 ]{};
-    real64 dDivMassFluxes_dFaceVars[ NF ]{};
-
-    // dof numbers
-    globalIndex const eqnRowLocalIndex = elemDofNumber[localIds[0]][localIds[1]][localIds[2]] - rankOffset;
-    globalIndex elemDofColIndices[ NF+1 ]{};
-    globalIndex faceDofColIndices[ NF ]{};
-    elemDofColIndices[0] = elemDofNumber[localIds[0]][localIds[1]][localIds[2]];
-
     // upwinded mobility
-    real64 upwMob = 0.0;
-    real64 dUpwMob_dp = 0.0;
+    real64 upwMobility = 0.0;
+    real64 dUpwMobility_dPres = 0.0;
     globalIndex upwDofNumber = 0;
 
     // for each element, loop over the one-sided faces
-    for( localIndex ifaceLoc = 0; ifaceLoc < NF; ++ifaceLoc )
+    for( integer ifaceLoc = 0; ifaceLoc < NUM_FACE; ++ifaceLoc )
     {
 
       // 1) Find if there is a neighbor, and if there is, grab the indices of the neighbor element
-      localIndex neighborIds[ 3 ] = { localIds[0], localIds[1], localIds[2] };
-      bool const hasNeighbor = hybridFVMKernels::CellConnectivity::findNeighbor( localIds,
-                                                                                 ifaceLoc,
-                                                                                 elemRegionList,
-                                                                                 elemSubRegionList,
-                                                                                 elemList,
-                                                                                 regionFilter,
-                                                                                 elemToFaces,
-                                                                                 neighborIds );
+      localIndex local[3] = { m_er, m_esr, ei };
+      localIndex neighbor[3] = { m_er, m_esr, ei };
+      bool const hasNeighbor =
+        hybridFVMKernels::CellConnectivity::
+          findNeighbor( local,
```

Comment:
So this  kernel should be renamed `isNeighborFound` or `hasNeighbor`?

Reply:
Done, I renamed this `bool` to `isNeighborFound`